static/panes/compiler.ts

```diff
@@ -0,0 +1,3561 @@
+import {GccSelectedPass} from './gccdump-view.interfaces';
+import {ToolInfo} from '../../lib/tooling/base-tool.interface';
+import {AssemblyInstructionInfo} from '../../lib/asm-docs/base';
+import {PPOptions} from './pp-view.interfaces';
+import {CompilationStatus} from '../compiler-service.interfaces';
+import {WidgetState} from '../widgets/libs-widget.interfaces';
+
+// @ts-ignore
+const toolIcons = require.context('../../views/resources/logos', false, /\.(png|svg)$/);
+
+const OpcodeCache = new LruCache({
+    max: 64 * 1024,
+    length: function (n) {
+        return JSON.stringify(n).length;
+    },
+});
+
+function patchOldFilters(filters) {
+    if (filters === undefined) return undefined;
+    // Filters are of the form {filter: true|false¸ ...}. In older versions, we used
+    // to suppress the {filter:false} form. This means we can't distinguish between
+    // "filter not on" and "filter not present". In the latter case we want to default
+    // the filter. In the former case we want the filter off. Filters now don't suppress
+    // but there are plenty of permalinks out there with no filters set at all. Here
+    // we manually set any missing filters to 'false' to recover the old behaviour of
+    // "if it's not here, it's off".
+    _.each(['binary', 'labels', 'directives', 'commentOnly', 'trim', 'intel'], function (oldFilter) {
+        if (filters[oldFilter] === undefined) filters[oldFilter] = false;
+    });
+    return filters;
+}
+
+const languages = options.languages;
+
+type LastResultType = {
+    ppOutput: unknown;
+    astOutput: unknown;
+    irOutput: unknown;
+    devices: unknown;
+    rustMirOutput: unknown;
+    rustMacroExpOutput: unknown;
+    rustHirOutput: unknown;
+    haskellCoreOutput: unknown;
+    haskellStgOutput: unknown;
+    haskellCmmOutput: unknown;
+    gccDumpOutput?: Record<GccDumpOptions, unknown>;
+    gnatDebugTreeOutput: unknown;
+    gnatDebugOutput: unknown;
+    optOutput: unknown;
+};
+
+type CurrentStateType = {
+    id?: number;
+    compiler: string;
+    source: number | null;
+    tree: any;
+    options: unknown;
+    filters: Record<string, unknown>;
+    wantOptInfo: any;
+    libs?: {name: string; ver: string}[];
+    lang: any;
+    selection: monaco.Selection | undefined;
+    flagsViewOpen: any;
+};
+
+type ActiveTools = {
+    id: number;
+    args: any;
+    stdin: string;
+};
+
+type NewToolSettings = {
+    toolId: number;
+    args: any;
+    stdin: string;
+};
+
+type LinkedCodeType = {
+    range: monaco.Range;
+    options: {
+        isWholeLine: boolean;
+        linesDecorationsClassName?: string;
+        className?: string;
+        inlineClassName?: string;
+    };
+};
+
+type Decorations = Record<string, monaco.editor.IModelDeltaDecoration[] | monaco.editor.IModelDeltaDecoration>;
+
+type CompileRequestOptions = {
+    userArguments: string;
+    compilerOptions: {
+        producePp: any;
```

Comment:
This `any` is ok for now

Reply:
Changed it to `PPOptions | null` anyways